src/OpenTelemetry.Instrumentation.Runtime/README.md

```diff
@@ -60,6 +60,138 @@ using var meterProvider = Sdk.CreateMeterProviderBuilder()
+
+Note: `ObservableGauge` should be changed to `ObservableUpDownCounter` once available,
+as `ObservableUpDownCounter` is the best fit of instrument type. The same applies
+to all the `ObservableGauge` below.
+
+The APIs used to retrieve the values are:
+
+* [GCMemoryInfo.TotalCommittedBytes](https://docs.microsoft.com/dotnet/api/system.gcmemoryinfo.totalcommittedbytes?view=net-6.0#system-gcmemoryinfo-totalcommittedbytes):
+  Gets the total committed bytes of the managed heap.
+
+* [GC.GetGCMemoryInfo().GenerationInfo[i].SizeAfterBytes](https://docs.microsoft.com/dotnet/api/system.gcgenerationinfo):
+  Represents the size in bytes of a generation on exit of the GC reported in GCMemoryInfo.
+
+* [GCGenerationInfo.FragmentationAfterBytes Property](https://docs.microsoft.com/dotnet/api/system.gcgenerationinfo.fragmentationafterbytes#system-gcgenerationinfo-fragmentationafterbytes)
+  Gets the fragmentation in bytes on exit from the reported collection.
+
+### JIT Compiler related metrics
+
+These metrics are only available when targeting .NET6 or later.
+
+| Name                                                  | Description                                                                                                                                                                                                                                                                                                                                  | Units       | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
+|-------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------|-------------------|------------|------------------|------------------|
+| process.runtime.dotnet.**jit.il_compiled.size**       | Count of bytes of intermediate language that have been compiled since the process start. The value will be zero under ahead-of-time (AOT) compilation mode.                                                                                                                                                                                  | `bytes`     | ObservableCounter | `Int64`    |                  |                  |
```

Comment:
```suggestion
| process.runtime.dotnet.**jit.il_compiled.size**       | Count of bytes of intermediate language that have been compiled since the process start.                                                                                                                                                                                  | `bytes`     | ObservableCounter | `Int64`    |                  |                  |
```

AOT compilation doesn't guarantee to pre-compile everything so the counter probably won't be zero even in that case.